### ISSUE TITLE
Intermittent `SchemaTests` failure

### ISSUE THREAD

**Issue by austinzheng** [realm/realm-cocoa#4585]
I occasionally see the following test failure on CI jobs running for PRs:

```
-[SchemaTests testMultipleProcessesTryingToInitializeSchema] : Asynchronous wait failed: Exceeded timeout of 10 seconds, with unfulfilled expectations: "notification fired".
```

Not sure what is causing it, why it's sporadic, etc, but filing a ticket to track.

Example:
https://ci.realm.io/job/objc_pr/configuration=Debug,swift_version=3.0.2,target=osx/4651/console

**Comment by jpsim** [comment]
Seen again in [this run](https://ci.realm.io/job/objc_pr/configuration=Debug,target=osx,xcode_version=8.0/4901/testReport/(root)/SchemaTests/testMultipleProcessesTryingToInitializeSchema/). @tgoyne would you object to disabling this test in debug configurations?

**Comment by austinzheng** [comment]
According to https://github.com/realm/realm-cocoa/pull/4789#pullrequestreview-29581582, https://github.com/realm/realm-core/pull/2552 may fix the underlying issue.

**Comment by jpsim** [comment]
Unfortunately saw this again in a PR job from #4755, which included core 2.6: https://ci.realm.io/job/objc_pr/5006/configuration=Debug,target=osx,xcode_version=8.0/
